File: frontend/src/app/service/comment.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { BASE_URL } from "../constant/base.url";
import { CommentInsertReqDto } from "../dto/comment/comment.insert.req.dto";
import { CommentListResDto } from "../dto/comment/comment.list.res.dto";

@Injectable({
    providedIn : 'root'
})
export class CommentService {

    constructor(private http : HttpClient) {}

    getAllIdTicket(id : number) : Observable<CommentListResDto> {
        return this.http.get<CommentListResDto>(`${BASE_URL.LOCALHOST}/comments/with_file_by_ticket/${id}`)
    }

    insert(data :CommentInsertReqDto) : Observable<CommentInsertReqDto>{
        return this.http.post<CommentInsertReqDto>(`${BASE_URL.LOCALHOST}/comments`, data);
    }

}

File: backend/src/main/java/com/lawencon/ticketms/service/CommentService.java
package com.lawencon.ticketms.service;

import com.lawencon.ticketms.dto.comment.CommentInsertReqDto;
import com.lawencon.ticketms.dto.comment.CommentInsertResDto;
import com.lawencon.ticketms.dto.comment.CommentListResDto;
import com.lawencon.ticketms.dto.comment.CommentResDto;

public interface CommentService {
	CommentInsertResDto insert(CommentInsertReqDto data);
	CommentListResDto getAllIdTicket(Long id);
	CommentResDto getById(Long id);
	CommentListResDto getAllIdCommentWithFile(Long id);
}
